pkgs/development/compilers/ghc/head.nix

```diff
@@ -22,7 +22,7 @@
 
 , # Whether to build dynamic libs for the standard library (on the target
   # platform). Static libs are always built.
-  enableShared ? true
+  enableShared ? !targetPlatform.useAndroidPrebuilt
```

Comment:
Can you apply the head.nix changes on more other GHCs for consistency?

Reply:
It's already applied to 8.4. No other GHC versions will be able to build for Android. It'd be better to have a "Not android" assertion in all the other GHCs. Want me to add this?

Reply:
Ah and there are fewer GHCs now too, +1.